src/main/java/com/mongodb/spark/sql/connector/read/MongoMicroBatchPartitionReader.java

```diff
@@ -0,0 +1,162 @@
+    this.readConfig = readConfig;
+    LOGGER.info(
+        "Creating partition reader for: PartitionId: {} with Schema: {}",
+        partition.getPartitionId(),
+        bsonDocumentToRowConverter.getSchema());
+  }
+
+  /** Proceed to next record, returns false if there is no more records. */
+  @Override
+  public boolean next() {
+    Assertions.ensureState(() -> !closed, () -> "Cannot call next() on a closed PartitionReader.");
+    MongoChangeStreamCursor<BsonDocument> cursor = getCursor();
+    BsonDocument next = cursor.tryNext();
+    if (next != null) {
+      if (readConfig.streamPublishFullDocumentOnly()) {
+        next = next.getDocument(FULL_DOCUMENT, new BsonDocument());
+      }
+      currentRow = bsonDocumentToRowConverter.toInternalRow(next);
+    }
+    return next != null;
+  }
+
+  /** Return the current record. This method should return same value until `next` is called. */
+  @Override
+  public InternalRow get() {
+    Assertions.ensureState(() -> !closed, () -> "Cannot call get() on a closed PartitionReader.");
+    return currentRow;
+  }
+
+  @Override
+  public void close() {
+    if (!closed) {
+      closed = true;
+      releaseCursor();
+    }
+  }
+
+  private MongoChangeStreamCursor<BsonDocument> getCursor() {
+    if (mongoClient == null) {
+      mongoClient = readConfig.getMongoClient();
+    }
+    if (changeStreamCursor == null) {
+
+      List<BsonDocument> pipeline = new ArrayList<>();
+      pipeline.add(
+          Aggregates.match(Filters.lt("clusterTime", partition.getEndOffsetTimestamp()))
+              .toBsonDocument());
+
+      if (readConfig.streamPublishFullDocumentOnly()) {
+        pipeline.add(Aggregates.match(Filters.exists(FULL_DOCUMENT)).toBsonDocument());
+      }
+      pipeline.addAll(partition.getPipeline());
+
+      ChangeStreamIterable<Document> changeStreamIterable =
+          mongoClient
+              .getDatabase(readConfig.getDatabaseName())
+              .getCollection(readConfig.getCollectionName())
+              .watch(pipeline)
+              .fullDocument(readConfig.getStreamFullDocument())
+              .startAtOperationTime(partition.getStartAtOperationTime());
+
+      try {
+        changeStreamCursor =
+            (MongoChangeStreamCursor<BsonDocument>)
+                changeStreamIterable.withDocumentClass(BsonDocument.class).cursor();
+        LOGGER.debug("Opened change stream cursor for partition: {}", partition);
+      } catch (RuntimeException e) {
+        throw new MongoSparkException("Could not create the change stream cursor.", e);
+      }
+    }
+    return changeStreamCursor;
+  }
+
+  private void releaseCursor() {
+    if (changeStreamCursor != null) {
+      LOGGER.debug("Closing cursor for partitionId: {}", partition.getPartitionId());
+      try {
+        changeStreamCursor.close();
+      } finally {
+        changeStreamCursor = null;
+        try {
+          mongoClient.close();
```

Comment:
We own the MongoClient as we called ReadConfig.getMongoClient().  In the background it uses the MongoClientCache which returns an instance of CachedMongoClient, that handles the delayed closure of MongoClients so that they can be reused if multiple tasks on the same worker need it.